### PR TITLE
Add cargo cache to CI

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -18,19 +18,26 @@ jobs:
...
         run:  rustup show
+
       - name: Check formatting of code
         run: cargo +stable fmt --check
-      - name: Clippy Checks
+
+      - name: Clippy checks
         run: cargo +stable clippy
 
   test:
```
Comment (fluidvanadium) [inline .github/workflows/ci.yaml:43]: name: Test
can this be more specific?

**Review by fluidvanadium — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Oscar-Pepper) [reply .github/workflows/ci.yaml:43]: hmm what do you suggest? unit-and-integration-test maybe?

Comment (Oscar-Pepper) [reply .github/workflows/ci.yaml:43]: or cargo-test ?

Comment (Oscar-Pepper) [reply .github/workflows/ci.yaml:43]: all our test jobs are not plural so `Cargo test` fits the format and also is the actual command so i think thats better

**Review by fluidvanadium — APPROVED** [review]

**Pull request merged**